### PR TITLE
ADM-383 T5B - [Backend] [Config]- Board section - Jira verify - part2

### PR DESCRIPTION
**Pull request by leeranzhi**
## Summary

...

## Before

_Description_

Before

## After

_Description_

After

## Note

_Null_

### CODE REVIEW

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 1]
```diff
@@ -1,50 +1,275 @@
 package heartbeat.service.board.jira;
 
+import java.util.*;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:3]: should not use `.*`

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 2]
```diff
@@ -1,50 +1,275 @@
...
+	public static final int QUERY_COUNT = 100;
+
 	private final JiraFeignClient jiraFeignClient;
 
+	private static final String DONE_CARD_TAG = "done";
+
+	public static final List<String> FIELDS_IGNORE = List.of("summary", "description", "attachment", "duedate",
+			"issuelinks");
+
+	@SuppressWarnings("PMD")
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:58]: update PMD.xml, and not add this here

**Review by guzhongren — APPROVED** [review]

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java (hunk 1 above)
Comment (leeranzhi) [reply backend/src/main/java/heartbeat/service/board/jira/JiraService.java]: Fixed.

File: backend/src/main/java/heartbeat/client/JiraFeignClient.java [hunk 3]
```diff
@@ -1,18 +1,44 @@
...
+import java.net.URI;
+import org.springframework.cache.annotation.Cacheable;
 import org.springframework.cloud.openfeign.FeignClient;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestHeader;
 
-import java.net.URI;
-
 @FeignClient(value = "jiraFeignClient", url = "https://demo.atlassian.net")
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/client/JiraFeignClient.java:15]: config it in `application.yaml`

**Review by guzhongren — COMMENTED** [review]
some change

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 4]
```diff
@@ -1,50 +1,333 @@
...
 		JiraBoardConfigDTO jiraBoardConfigDTO;
 		try {
-			jiraBoardConfigDTO = jiraFeignClient.getJiraBoardConfiguration(URI.create(url), boardRequest.getBoardId(),
-					boardRequest.getToken());
-			return BoardConfigResponse.builder()
-				.jiraColumns(jiraBoardConfigDTO.getColumnConfig()
-					.getColumns()
-					.stream()
-					.map(this::mapToResponseColumn)
+			log.info("[Jira] Start to get configuration for board, name: "
```
[the remark this answers is not in the document]
Comment (leeranzhi) [reply backend/src/main/java/heartbeat/service/board/jira/JiraService.java]: fixed.

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 5]
```diff
@@ -1,50 +1,333 @@
...
+				+ boardRequestParam.getBoardName());
+			jiraBoardConfigDTO = jiraFeignClient.getJiraBoardConfiguration(baseUrl,
+				boardRequestParam.getBoardId(),
+				boardRequestParam.getToken());
+			log.info("[Jira] Successfully get configuration_data, name: {}, id: {}",
```
[the remark this answers is not in the document]
Comment (leeranzhi) [reply backend/src/main/java/heartbeat/service/board/jira/JiraService.java]: fixed.

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java (hunk 2 above)
Comment (leeranzhi) [reply backend/src/main/java/heartbeat/service/board/jira/JiraService.java]: fixed.

File: backend/src/main/java/heartbeat/client/JiraFeignClient.java (hunk 3 above)
Comment (leeranzhi) [reply backend/src/main/java/heartbeat/client/JiraFeignClient.java]: fixed.

**Review by Teresasasa — APPROVED** [review]
LGTM

**Pull request merged**